datalad_metalad/aggregate.py

```diff
@@ -445,10 +445,44 @@ def __call__(
                 into))
 
 
+def _extractor_changed(ds, ds_db, exinfo, path):
+    """Determine whether or not there's a change in the extractors
+    between recorded metadata in `ds_db` and `exinfo`
+
+    Parameters
+    ----------
+
+    ds: Dataset
+      dataset with recorded metadata on `path`
+    ds_db: dict
+      aggregation info of `ds`
+    exinfo: dict
+      extractor configuration to compare to
+    path: Path
+      thing we want metadata about
+    """
+
+    exstate_rec = ds_db.get(path, {}).get('extractors', None)
+
+    # check recorded states in src and dst vs. "current"
+    if (
+            # old aggregate catalog with a plain extractor name list
+            not isinstance(exstate_rec, dict) \
+            or sorted(exinfo.keys()) != sorted(exstate_rec.keys()) \
+            or any(exinfo[k]['state'] != exstate_rec[k]
+                   for k in exinfo)
+         ):
+
+        lgr.debug('Difference between recorded and current extractor detected in %s (was: %s; is: %s)',
+                  ds, exstate_rec, exinfo.get('state', None))
```

Comment:
very minor, but this leads to a confusing Debug message. I believe ``exinfo`` doesn't have key ``state`` (only the subdictionaries of any extractor have it). I had the message 

```
'Difference between recorded and current extractor detected in tmp/mydataset (was: None; is: None)'
``` 
when trying to get aggregated subds metadata from subds into a toplevel dataset with no metadata record of the subds.
I guess ``exinfo.get('extractors', None)`` could show a difference in extractors between the two dataset infos, but it wouldn't show a difference in case of 
```py
 any(exinfo[k]['state'] != exstate_rec[k]
                   for k in exinfo)
```